src/code/providers/interactive-api-provider.ts

```diff
@@ -118,10 +125,10 @@ class InteractiveApiProvider extends ProviderInterface {
     }
   }
 
-  async readAttachmentContent() {
-    const response = await readAttachment(kAttachmentFilename)
+  async readAttachmentContent(interactiveState: InteractiveStateAttachment) {
+    const response = await readAttachment(interactiveState.__attachment__)
     if (response.ok) {
-      return response.text()
+      return interactiveState.contentType === "application/json" ? response.json() : response.text()
```

Comment:
The response should have a content type header that ought to be a safer way to determine the content type instead of looking at the interactive state.  The state could possibly get out of sync with the attachment, but the content type header should always match actual attachment file.

There might be an issue with the setup of the CORS for the S3 bucket so the browser is allowed to read the content type header in the response. 

I'm fine with punting this change so that you can move on.